### PR TITLE
[PB-4731]: implement caching for user avatars in cache manager

### PR DESCRIPTION
**Pull request by terrerox**
- Added methods to CacheManagerService for getting, setting, and deleting user avatars.
- Updated UserController to use cached avatar retrieval.
- Refactored user use cases to include getCachedAvatar logic, handling cache hits and misses.
- Enhanced unit tests to cover new avatar caching functionality and edge cases.

### CODE REVIEW

**Comment by terrerox** [comment]
> @terrerox What does this fix?

@apsantiso We treat `avatar/refresh` as the source of truth. Even if a cached (non-expired) URL exists, the client always calls avatar/refresh first. We cache the result to IndexedDB for performance, but since IndexedDB is per-origin, another browser/origin (e.g., localhost:3000) could delete the image and leave a stale URL behind. Using `avatar/refresh` ensures we don’t show an invalid URL if the avatar was removed elsewhere.

File: src/modules/cache-manager/cache-manager.service.ts [hunk 1]
```diff
@@ -7,7 +7,9 @@ export class CacheManagerService {
   private readonly USAGE_KEY_PREFIX = 'usage:';
   private readonly LIMIT_KEY_PREFIX = 'limit:';
   private readonly JWT_KEY_PREFIX = 'jwt:';
+  private readonly AVATAR_KEY_PREFIX = 'avatar:';
   private readonly TTL_10_MINUTES = 10000 * 60;
+  private readonly TTL_24_HOURS = 24 * 60 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (apsantiso) [reply src/modules/cache-manager/cache-manager.service.ts:12]: Nvm, i see it is set to 24 h.

**Review by apsantiso — APPROVED** [review]
